sympy/printing/theanocode.py

```diff
@@ -6,6 +6,13 @@
 import sympy
 from functools import partial
 
+from sympy.utilities.exceptions import SymPyDeprecationWarning
+
+SymPyDeprecationWarning(
+    feature="sympy.printing.theanocode",
+    useinstead="Theano is deprecated; use Aesara and sympy.printing.aesaracode",
+    deprecated_since_version="1.6").warn()
```

Comment:
Adding a DeprecationWaring on import has turned out to be annoying in the past. Is there instead some function call where the warning could be emitted?

For tests there is the `warns_deprecated_sympy` context manager that can be used to assert that a warning is emitted while testing code.

The `deprecated_since_version` should be 1.8 (the next release of sympy).

There should be an `issue=1234` argument giving the issue number of a deprecation issue on the sympy/sympy github repo. The deprecation issue should be created first (before this is merged) so that the issue number can be added in this PR. Users will be presented with a link to the issue when seeing the warning. The issue should explain what is deprecated and why and how users can update their code and/or write code that spans old and new versions if relevant. The issue should also explain what should be done when the deprecation period is over (e.g. "delete files x and y, remove class z"). Potentially users would use that issue to ask questions about the deprecation.

Reply:
I've created https://github.com/sympy/sympy/issues/21150 for this purpose.